File: Operations/Classes/Employee.cs
using System;

namespace Operations.Classes
{
    public class Employee : IEquatable<Employee>
    {
        public int EmployeeIdentifier { get;set;}
        public string FirstName {get;set;}
        public string LastName {get;set;} 
        public string DepartmentName {get;set;}
        public bool Equals(Employee other)
        {
            return other.EmployeeIdentifier == EmployeeIdentifier && 
                   other.FirstName == FirstName && 
                   other.DepartmentName == DepartmentName;

        }

        public override int GetHashCode()
        {
            return StringComparer.OrdinalIgnoreCase.GetHashCode(FirstName) ^
                   StringComparer.OrdinalIgnoreCase.GetHashCode(LastName) ^
                   StringComparer.OrdinalIgnoreCase.GetHashCode(DepartmentName);
        }

        public override string ToString()
        {
            return base.ToString();
        }
    }
}

File: Operations_vb/Classes/Employee.vb
Imports Extensions

Namespace Classes
    Public Class Employee
        Implements IEquatable(Of Employee)

        Public Property EmployeeIdentifier() As Integer
        Public Property FirstName() As String
        Public Property LastName() As String
        Public Property DepartmentName() As String
        Public Overloads Function Equals(other As Employee) As Boolean _
            Implements IEquatable(Of Employee).Equals

            Return other.FirstName.AreEqual(FirstName) AndAlso
                   other.LastName.AreEqual(LastName) AndAlso
                   other.DepartmentName.AreEqual(DepartmentName)

        End Function

        Public Overrides Function GetHashCode() As Integer

            Return StringComparer.OrdinalIgnoreCase.GetHashCode(FirstName) Xor
                   StringComparer.OrdinalIgnoreCase.GetHashCode(LastName) Xor
                   StringComparer.OrdinalIgnoreCase.GetHashCode(DepartmentName)

        End Function

        Public Overrides Function ToString() As String
            Return MyBase.ToString()
        End Function
    End Class
End Namespace
